Bump version 0.10.0

benbovy opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

benbovy committed:
bump v0.10.0

benbovy committed:
abseil is now an external dependency

benbovy committed:
reset build number

benbovy committed:
build with C++17 standard

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

benbovy commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.22.0, conda-smithy 3.21.2, and conda-forge-pinning 2022.10.19.09.53.59

benbovy committed:
remove all patches

benbovy committed:
re-add and update cmake patch (gtest)

S2Geometry has no support for automatically fetching and building gtest

benbovy committed:
override cpp standard

benbovy committed:
Use C++11 standard on windows

https://github.com/conda-forge/abseil-cpp-feedstock/issues/45#issuecomment-1236384709

Or should we only depend on libabseil here?

benbovy committed:
probably only need libabseil (also at runtime)

benbovy committed:
win: s2.dll now installed in the right location

Apparently...

benbovy committed:
Try fix PRId64 gcc error on linux

https://stackoverflow.com/questions/52715250/strange-compiler-behavior-with-priu64-in-c

I'd expected it to work without it for any recent version gcc, though.

benbovy committed:
try fix gcc PRId64 on linux 2

benbovy committed:
try fix gcc PRId64 error on linux 3

benbovy committed:
try fix gcc PRId64 format error on linux 4

benbovy committed:
try fix gcc PRId64 on linux good one

Not sure why the CMake patches in the previous commits didn't work.
Patch the problematic source files instead.
